plugins/inputs/aliyuncms/discovery.go

```diff
@@ -379,21 +369,21 @@ func (dt *discoveryTool) getDiscoveryData(cli aliyunSdkClient, req *requests.Com
 			preparedData := map[string]interface{}{}
 
 			for _, raw := range discoveryData {
-				if elem, ok := raw.(map[string]interface{}); ok {
-					if objectID, ok := elem[dt.respObjectIDKey].(string); ok {
-						preparedData[objectID] = elem
-					}
-				} else {
-					return nil, errors.Errorf("Can't parse input data element, not a map[string]interface{} type")
+				elem, ok := raw.(map[string]interface{})
+				if !ok {
+					return nil, errors.Errorf("can't parse input data element, not a map[string]interface{} type")
+				}
+				if objectID, ok := elem[dt.respObjectIDKey].(string); ok {
+					preparedData[objectID] = elem
 				}
 			}
-
 			return preparedData, nil
 		}
 	}
 }
 
-func (dt *discoveryTool) getDiscoveryDataAllRegions(limiter chan bool) (map[string]interface{}, error) {
+// GetDiscoveryDataAcrossRegions returns discovery data across defined (when creating) list of regions
+func (dt *discoveryTool) GetDiscoveryDataAcrossRegions(lmtr chan bool) (map[string]interface{}, error) {
```

Comment:
Please remove the commented line and unexport `GetDiscoveryDataAcrossRegions()`.

Reply:
For the sake of consistency I will switch all functions of `discoveryTool` object + `NewDiscoveryTool` function to be un-exported, as discussed.

Reply:
Done

Reply:
Done